src/main/scala/com/typesafe/sbt/bundle/LagomBundle.scala

```diff
@@ -185,17 +185,45 @@ object LagomBundle extends AutoPlugin {
   private def toClasspathUrls(attributedFiles: Seq[Attributed[File]]): Array[URL] =
     attributedFiles.files.map(_.toURI.toURL).toArray
 
+  private val pathBeginExtractor = """^\\Q(/.*)\\E.*""".r
+
   /**
     * Convert services string to `Map[String, Endpoint]` by using the Play json library
     */
   private def toConductrEndpoints(services: String, servicePort: Int): Map[String, Endpoint] = {
     val json = Json.parse(services)
-    val serviceNames = (json \\ "name").flatMap(_.asOpt[String])
-    val formattedServiceNames = serviceNames.map {
-      case name if name.startsWith("/") => name.drop(1) // TODO: The drop will not be necessary anymore once Lagom verifies the service name
-      case name => name
+    val serviceNamesAndPaths = json.as[List[JsObject]].map { o =>
+      val serviceName = (o \ "name").as[String]
+      val pathlessServiceName = if (serviceName.startsWith("/")) serviceName.drop(1) else serviceName
+      val pathBegins = (o \ "acls" \\ "pathPattern")
+        .map(_.as[String])
+        .collect {
+          case pathBeginExtractor(pathBegin) =>
+            (if (pathBegin.endsWith("/")) pathBegin.dropRight(1) else pathBegin) + "?preservePath"
+        }
+      pathlessServiceName -> pathBegins
     }
-    formattedServiceNames.map(name => name -> Endpoint("http", 0, Set(URI(s"http://:$servicePort/$name")))).toMap
+    serviceNamesAndPaths
```

Comment:
This logic is pretty hard to understand. It should be more readable if we create multiple functions for that (one for the fold block, foldLeft block, etc.)